### PR TITLE
refactor: close resource in ZipFolder.java

### PR DESCRIPTION
**Pull request by zielint0**
Make sure a resource is properly released.

@pvojtechovsky 
Pavel could you take a look on *public List<SpoonFile> getFiles()* and tell me if this algorithm is correct *before* and *after* change.

### CODE REVIEW

**Comment by monperrus** [comment]
LGTM. @pvojtechovsky do you merge?

File: src/main/java/spoon/support/compiler/ZipFolder.java [hunk 1]
```diff
@@ -74,22 +74,19 @@ public ZipFolder(File file) throws IOException {
...
 			files = new ArrayList<>();
-			try (ZipInputStream zipInput = new ZipInputStream(new BufferedInputStream(new FileInputStream(file)))) {
+			final int buffer = 2048;
+			try (ZipInputStream zipInput = new ZipInputStream(new BufferedInputStream(new FileInputStream(file)));
+				ByteArrayOutputStream output = new ByteArrayOutputStream(buffer)) {
 				ZipEntry entry;
 				while ((entry = zipInput.getNextEntry()) != null) {
 					// deflate in buffer
-					final int buffer = 2048;
-					ByteArrayOutputStream output = new ByteArrayOutputStream(buffer);
```
[the remark this answers is not in the document]
Comment (zielint0) [reply src/main/java/spoon/support/compiler/ZipFolder.java]: Thank you Pavel. I felt there is something wrong with it.

**Comment by pvojtechovsky** [comment]
@monperrus please review my changes too and merge it - if ok. Thank you.

**Comment by monperrus** [comment]
Thanks for the added test, a new test is always valuable.

**Pull request merged**